robot/EDA/resources/EDA.py

```diff
@@ -481,26 +483,51 @@ def choose_panel_tab(self,tab):
             error="Cannot find the panel tab: " + tab
         )
         self.selenium.click_element(eda_lex_locators["choose_tab"].format(tab))
-
-    def select_tab(self, tab):
-        """ Select tab - as passed in to the function
-            tab is a label on menu item, and not a locator.
+        
+        
+    @capture_screenshot_on_error    
+    def select_tab(self, title):
+        """ Switch between different tabs on a record page like Related, Details, News, Activity and Chatter
+            Pass title of the tab
         """
-        locator_menu = eda_lex_locators["tab_menu"].format(tab)
-        element_menu = self.selenium.driver.find_element_by_xpath(locator_menu)
-        locator_tab = eda_lex_locators["tab_tab"].format(tab)
-        self.selenium.wait_until_page_contains_element(
-            locator_menu,
-            error="Contact tab unavailable"
-        )
-        # javascript is being used here because the usual selenium click is highly unstable for this element on MetaCI
-        self.selenium.driver.execute_script("arguments[0].click()", element_menu)
-        time.sleep(1)
-
-        # Sometimes, single click fails. Hence an additional condition to click on it again
-        if not self.check_if_element_exists(locator_tab):
-            self.selenium.driver.execute_script("arguments[0].click()", element_menu)
-            time.sleep(1)
+        tab_found = False
+        locators = eda_lex_locators["tabs"].values()
+        for i in locators:
+            locator = i.format(title)
+            if self.check_if_element_exists(locator):
+                print(locator)
+                buttons = self.selenium.get_webelements(locator)
+                for button in buttons:
+                    print(button)
+                    if button.is_displayed():
+                        print("button displayed is {}".format(button))
+                        self.salesforce._focus(button)
+                        button.click()
+                        time.sleep(5)
```

Comment:
Is this waiting for something to load?

Reply:
yes as I explained to you in the meeting, not putting it here